Content.Shared/Nutrition/EntitySystems/HungerSystem.cs

```diff
@@ -120,7 +151,7 @@ private void UpdateCurrentThreshold(EntityUid uid, HungerComponent? component =
             return;
         component.CurrentThreshold = calculatedHungerThreshold;
         DoHungerThresholdEffects(uid, component);
-        Dirty(uid, component);
+        SetAuthoritativeHungerValue((uid, component), GetHunger(component));
```

Comment:
Why is this necessary?

Reply:
My thinking is any time the threshold changes, the "start point" values should be updated as well.
I suppose that's not strictly true as the only change that necessitates an update is a change to the decay rate. So I'll move this call down to exactly where the decay rate is modified.